src/state/opamSysInteract.ml

```diff
@@ -460,8 +460,14 @@ module Cygwin = struct
   (* Set setup.exe in the good place, ie in .opam/.cygwin/ *)
   let check_setup ~update =
     let dst = cygsetup () in
-    if update || not (OpamFilename.exists dst) then
-      OpamProcess.Job.run @@ download_setupexe dst
+    let download () = download_setupexe dst in
```

Comment:
The fallback is added in `check_setup` and not in `donwload_setup_exe`.
There is another use of `donwload_setup_exe` in Cygwin internal `install` function. It is mainly called via an init, but it can be also called via a reinit. That one could fail too.